netbox_4_3: init at 4.3.2

transcaffeine opened the pull request:
## Things done

- Add `netbox_4_3` package
  - Update python dependencies `strawberry-graphql` and `strawberry-django`
- Adjust nixos (upgrade) tests for `netbox_4_3`
- drop the `netbox_3_7` upgrade test, apparently a left-over from https://github.com/NixOS/nixpkgs/pull/401110

Attempting to continue this from https://github.com/NixOS/nixpkgs/pull/405880#issuecomment-2940757749

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

transcaffeine committed:
netbox_4_3: init at 4.3.2

Release notes:
  - https://github.com/netbox-community/netbox/releases/tag/v4.3.0
  - https://github.com/netbox-community/netbox/releases/tag/v4.3.1
  - https://github.com/netbox-community/netbox/releases/tag/v4.3.2
Full changelog: https://github.com/netbox-community/netbox/compare/v4.2.9...v4.3.2

transcaffeine committed:
nixos/tests/netbox: Adjust (upgrade) test for 4.3.0+, drop tests for 3.7

transcaffeine committed:
python3Packages.strawberry-graphql: 0.263.1 -> 0.271.0

transcaffeine committed:
python3Packages.strawberry-django: 0.57.1 -> 0.60.0

dotlambda commented on pkgs/by-name/ne/netbox_4_3/package.nix:
```diff
@@ -0,0 +1,130 @@
+  nix-update-script,
+}:
+let
+  py = python3.override {
+    packageOverrides = _final: prev: { django = prev.django_5; };
```
```suggestion
  py = python3.override {
    self = py;
    packageOverrides = _final: prev: { django = prev.django_5; };
```

dotlambda commented on pkgs/by-name/ne/netbox_4_3/package.nix:
```diff
@@ -0,0 +1,130 @@
+py.pkgs.buildPythonApplication rec {
+  pname = "netbox";
+  version = "4.3.2";
+
+  format = "other";
```
```suggestion
  pyproject = false;
```

dotlambda commented on pkgs/by-name/ne/netbox_4_3/package.nix:
```diff
@@ -0,0 +1,130 @@
+    ./custom-static-root.patch
+  ];
+
+  propagatedBuildInputs =
```
```suggestion
  dependencies =
```

dotlambda commented on pkgs/by-name/ne/netbox_4_3/package.nix:
```diff
@@ -0,0 +1,130 @@
+        nh3
+        pillow
+        psycopg
+        psycopg.optional-dependencies.c
+        psycopg.optional-dependencies.pool
```
Use `++`

dotlambda commented on pkgs/by-name/ne/netbox_4_3/package.nix:
```diff
@@ -0,0 +1,130 @@
+        # for error reporting
+        sentry-sdk
+      ]
+      ++ social-auth-core.passthru.optional-dependencies.openidconnect
```
```suggestion
      ++ social-auth-core.optional-dependencies.openidconnect
```

dotlambda commented on pkgs/by-name/ne/netbox_4_3/package.nix:
```diff
@@ -0,0 +1,130 @@
+    )
+    ++ extraBuildInputs;
+
+  buildInputs = with py.pkgs; [
```
Is this supposed to be `nativeBuildInputs`?

dotlambda commented on pkgs/by-name/ne/netbox_4_3/package.nix:
```diff
@@ -0,0 +1,130 @@
+
+  postBuild = ''
+    PYTHONPATH=$PYTHONPATH:netbox/
+    python -m mkdocs build
```
```suggestion
    ${py.interpreter} -m mkdocs build
```

dotlambda commented on pkgs/by-name/ne/netbox_4_3/package.nix:
```diff
@@ -0,0 +1,130 @@
+  '';
+
+  passthru = {
+    python = python3;
```
```suggestion
    python = py;
```

dotlambda commented on pkgs/by-name/ne/netbox_4_3/package.nix:
```diff
@@ -0,0 +1,130 @@
+  };
+
+  meta = {
+    homepage = "https://github.com/netbox-community/netbox";
```
```suggestion
    changelog = "https://github.com/netbox-community/netbox/blob/${src.tag}/docs/release-notes/version-${lib.versions.majorMinor version}.md";
    homepage = "https://github.com/netbox-community/netbox";
```

dotlambda commented on pkgs/development/python-modules/strawberry-django/default.nix:
```diff
@@ -33,14 +34,14 @@
+    hash = "sha256-mMI/tPdt9XK6Lz7VmI3uDxcCjIuidUeGHjG+6AQLoeQ=";
   };
 
   build-system = [
```
Can you ask upstream to remove setuptools from this? Or is it really needed?

leona-ya commented on nixos/modules/services/web-apps/netbox.nix:
```diff
@@ -103,19 +103,17 @@ in
       type = lib.types.package;
       default =
         if lib.versionAtLeast config.system.stateVersion "25.05" then
```
This is inconsitent with the defaultText, and here it's wrong: Here, `25.11` should be used.

minijackson reviewed (commented):
Thanks!

Can you mark NetBox 4.2 as EOL? Same as here: https://github.com/NixOS/nixpkgs/blob/7a2c2ffb6b059f4800aca8d97b846103808fb635/pkgs/by-name/ne/netbox_4_1/package.nix#L130-L132

Some plugins fail to build, I think because of a wrong `disabled` attribute. You should be able to replace it like so:

``` diff
-  disabled = pythonAtLeast "3.13";
+  disabled = python.pythonVersion != netbox.python.pythonVersion;
```

After that, I think this PR just needs a rebase to fix the merge conflict.

---

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 414367`
Commit: `10237aa70c98f09f66d3bd9dc23fe6bf413a4fc4`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>netbox_4_1</li>
  </ul>
</details>
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:x: 10 packages failed to build:</summary>
  <ul>
    <li>python312Packages.netbox-attachments</li>
    <li>python312Packages.netbox-attachments.dist</li>
    <li>python312Packages.netbox-contract</li>
    <li>python312Packages.netbox-contract.dist</li>
    <li>python312Packages.netbox-floorplan-plugin</li>
    <li>python312Packages.netbox-floorplan-plugin.dist</li>
    <li>python312Packages.netbox-napalm-plugin</li>
    <li>python312Packages.netbox-napalm-plugin.dist</li>
    <li>python312Packages.netbox-topology-views</li>
    <li>python312Packages.netbox-topology-views.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 39 packages built:</summary>
  <ul>
    <li>netbox (netbox_4_3)</li>
    <li>netbox_4_2</li>
    <li>nixpkgs-manual</li>
    <li>python312Packages.netbox-bgp</li>
    <li>python312Packages.netbox-bgp.dist</li>
    <li>python312Packages.netbox-documents</li>
    <li>python312Packages.netbox-documents.dist</li>
    <li>python312Packages.netbox-interface-synchronization</li>
    <li>python312Packages.netbox-interface-synchronization.dist</li>
    <li>python312Packages.netbox-plugin-prometheus-sd</li>
    <li>python312Packages.netbox-plugin-prometheus-sd.dist</li>
    <li>python312Packages.netbox-qrcode</li>
    <li>python312Packages.netbox-qrcode.dist</li>
    <li>python312Packages.netbox-reorder-rack</li>
    <li>python312Packages.netbox-reorder-rack.dist</li>
    <li>python312Packages.netbox-routing</li>
    <li>python312Packages.netbox-routing.dist</li>
    <li>python312Packages.strawberry-django</li>
    <li>python312Packages.strawberry-django.dist</li>
    <li>python312Packages.strawberry-graphql</li>
    <li>python312Packages.strawberry-graphql.dist</li>
    <li>python313Packages.netbox-bgp</li>
    <li>python313Packages.netbox-bgp.dist</li>
    <li>python313Packages.netbox-documents</li>
    <li>python313Packages.netbox-documents.dist</li>
    <li>python313Packages.netbox-interface-synchronization</li>
    <li>python313Packages.netbox-interface-synchronization.dist</li>
    <li>python313Packages.netbox-plugin-prometheus-sd</li>
    <li>python313Packages.netbox-plugin-prometheus-sd.dist</li>
    <li>python313Packages.netbox-qrcode</li>
    <li>python313Packages.netbox-qrcode.dist</li>
    <li>python313Packages.netbox-reorder-rack</li>
    <li>python313Packages.netbox-reorder-rack.dist</li>
    <li>python313Packages.netbox-routing</li>
    <li>python313Packages.netbox-routing.dist</li>
    <li>python313Packages.strawberry-django</li>
    <li>python313Packages.strawberry-django.dist</li>
    <li>python313Packages.strawberry-graphql</li>
    <li>python313Packages.strawberry-graphql.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.netbox-attachments</summary>
<pre>@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: netbox_attachments
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <lambda>
  File "/nix/store/89laaz8x9pdydxzz74p0vd8vdj0szqbq-python3-3.12.11/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 999, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/nix/store/ncc1nxg8hixwkvvf75dzpdz2vgvaag2m-python3.12-netbox-attachments-7.2.0/lib/python3.12/site-packages/netbox_attachments/__init__.py", line 1, in <module>
    from netbox.plugins import PluginConfig
  File "/nix/store/prhfcmf1nn58xs152fwcxkxs6biv5pgc-netbox-4.3.2/opt/netbox/netbox/netbox/plugins/__init__.py", line 4, in <module>
    from django.apps import AppConfig
ModuleNotFoundError: No module named 'django'</pre>
</details>
<details>
<summary>python312Packages.netbox-contract</summary>
<pre>Check whether the following modules can be imported: netbox_contract
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <lambda>
  File "/nix/store/89laaz8x9pdydxzz74p0vd8vdj0szqbq-python3-3.12.11/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 999, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/nix/store/gxahcxhzwipwaf136g2zb9vqsxhdp5zz-python3.12-netbox-contract-2.3.2/lib/python3.12/site-packages/netbox_contract/__init__.py", line 1, in <module>
    from netbox.plugins import PluginConfig
  File "/nix/store/prhfcmf1nn58xs152fwcxkxs6biv5pgc-netbox-4.3.2/opt/netbox/netbox/netbox/plugins/__init__.py", line 11, in <module>
    from netbox.search import register_search
  File "/nix/store/prhfcmf1nn58xs152fwcxkxs6biv5pgc-netbox-4.3.2/opt/netbox/netbox/netbox/search/__init__.py", line 6, in <module>
    from netaddr import IPAddress, IPNetwork
ModuleNotFoundError: No module named 'netaddr'</pre>
</details>
<details>
<summary>python312Packages.netbox-floorplan-plugin</summary>
<pre>@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: netbox_floorplan
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <lambda>
  File "/nix/store/89laaz8x9pdydxzz74p0vd8vdj0szqbq-python3-3.12.11/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 999, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/nix/store/zc5pgk39lz5d14a0wfvw72vj9w1201l2-python3.12-netbox-floorplan-plugin-0.6.0/lib/python3.12/site-packages/netbox_floorplan/__init__.py", line 1, in <module>
    from netbox.plugins import PluginConfig
  File "/nix/store/prhfcmf1nn58xs152fwcxkxs6biv5pgc-netbox-4.3.2/opt/netbox/netbox/netbox/plugins/__init__.py", line 4, in <module>
    from django.apps import AppConfig
ModuleNotFoundError: No module named 'django'</pre>
</details>
<details>
<summary>python312Packages.netbox-napalm-plugin</summary>
<pre>@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: netbox_napalm_plugin
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <lambda>
  File "/nix/store/89laaz8x9pdydxzz74p0vd8vdj0szqbq-python3-3.12.11/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 999, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/nix/store/4cjs4c8p9jrax036q1mgkd0hzzdfz949-python3.12-netbox-napalm-plugin-0.3.1/lib/python3.12/site-packages/netbox_napalm_plugin/__init__.py", line 8, in <module>
    from netbox.plugins import PluginConfig
  File "/nix/store/prhfcmf1nn58xs152fwcxkxs6biv5pgc-netbox-4.3.2/opt/netbox/netbox/netbox/plugins/__init__.py", line 4, in <module>
    from django.apps import AppConfig
ModuleNotFoundError: No module named 'django'</pre>
</details>
<details>
<summary>python312Packages.netbox-topology-views</summary>
<pre>@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: netbox_topology_views
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <lambda>
  File "/nix/store/89laaz8x9pdydxzz74p0vd8vdj0szqbq-python3-3.12.11/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 999, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/nix/store/lqglh0z7viwgfqpf0jvkf468wvr7a8yi-python3.12-netbox-topology-views-4.2.1/lib/python3.12/site-packages/netbox_topology_views/__init__.py", line 1, in <module>
    from netbox.plugins import PluginConfig
  File "/nix/store/prhfcmf1nn58xs152fwcxkxs6biv5pgc-netbox-4.3.2/opt/netbox/netbox/netbox/plugins/__init__.py", line 4, in <module>
    from django.apps import AppConfig
ModuleNotFoundError: No module named 'django'</pre>
</details>
</details>

transcaffeine committed:
netbox_4_2: mark as EOL

transcaffeine committed:
python312Packages.netbox-attachments: only build for python version of netbox

transcaffeine committed:
python312Packages.netbox-contract: only build for python version of netbox

transcaffeine committed:
python312Packages.netbox-floorplan-plugin: only build for python version of netbox

transcaffeine committed:
python312Packages.netbox-napalm-plugin: only build for python version of netbox

transcaffeine committed:
python312Packages.netbox-topology-views: only build for python version of netbox

transcaffeine committed:
python313Packages.netbox-contract: 2.3.2 -> 2.4.0

Release notes: https://github.com/mlebreuil/netbox-contract/releases/tag/v2.4.0
Full changelog: https://github.com/mlebreuil/netbox-contract/compare/v2.3.2...v2.4.0

transcaffeine committed:
python313Packages.netbox-attachments: 7.2.0 -> 8.0.4

Release notes: https://github.com/Kani999/netbox-attachments/releases/tag/8.0.4
Full changelog: https://github.com/Kani999/netbox-attachments/compare/7.2.0...8.0.4

transcaffeine committed:
python313Packages.netbox-floorplan-plugin: 0.6.0 -> 0.7.0

Release notes: https://github.com/netbox-community/netbox-floorplan-plugin/releases/tag/0.7.0
Full changelog: https://github.com/netbox-community/netbox-floorplan-plugin/compare/0.6.0...0.7.0

transcaffeine committed:
python313Packages.netbox-topology-views: 4.2.1 -> 4.3.0

Release notes: https://github.com/netbox-community/netbox-topology-views/releases/tag/v4.3.0
Full changelog: https://github.com/netbox-community/netbox-topology-views/compare/v4.2.1...v4.3.0

transcaffeine commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 414367`
Commit: `4bfb13e0f4043757ca517ed276aa56f6983c3e5a`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 48 packages built:</summary>
  <ul>
    <li>netbox</li>
    <li>nixpkgs-manual</li>
    <li>python312Packages.netbox-bgp</li>
    <li>python312Packages.netbox-bgp.dist</li>
    <li>python312Packages.netbox-documents</li>
    <li>python312Packages.netbox-documents.dist</li>
    <li>python312Packages.netbox-interface-synchronization</li>
    <li>python312Packages.netbox-interface-synchronization.dist</li>
    <li>python312Packages.netbox-plugin-prometheus-sd</li>
    <li>python312Packages.netbox-plugin-prometheus-sd.dist</li>
    <li>python312Packages.netbox-qrcode</li>
    <li>python312Packages.netbox-qrcode.dist</li>
    <li>python312Packages.netbox-reorder-rack</li>
    <li>python312Packages.netbox-reorder-rack.dist</li>
    <li>python312Packages.netbox-routing</li>
    <li>python312Packages.netbox-routing.dist</li>
    <li>python312Packages.strawberry-django</li>
    <li>python312Packages.strawberry-django.dist</li>
    <li>python312Packages.strawberry-graphql</li>
    <li>python312Packages.strawberry-graphql.dist</li>
    <li>python313Packages.netbox-attachments</li>
    <li>python313Packages.netbox-attachments.dist</li>
    <li>python313Packages.netbox-bgp</li>
    <li>python313Packages.netbox-bgp.dist</li>
    <li>python313Packages.netbox-contract</li>
    <li>python313Packages.netbox-contract.dist</li>
    <li>python313Packages.netbox-documents</li>
    <li>python313Packages.netbox-documents.dist</li>
    <li>python313Packages.netbox-floorplan-plugin</li>
    <li>python313Packages.netbox-floorplan-plugin.dist</li>
    <li>python313Packages.netbox-interface-synchronization</li>
    <li>python313Packages.netbox-interface-synchronization.dist</li>
    <li>python313Packages.netbox-napalm-plugin</li>
    <li>python313Packages.netbox-napalm-plugin.dist</li>
    <li>python313Packages.netbox-plugin-prometheus-sd</li>
    <li>python313Packages.netbox-plugin-prometheus-sd.dist</li>
    <li>python313Packages.netbox-qrcode</li>
    <li>python313Packages.netbox-qrcode.dist</li>
    <li>python313Packages.netbox-reorder-rack</li>
    <li>python313Packages.netbox-reorder-rack.dist</li>
    <li>python313Packages.netbox-routing</li>
    <li>python313Packages.netbox-routing.dist</li>
    <li>python313Packages.netbox-topology-views</li>
    <li>python313Packages.netbox-topology-views.dist</li>
    <li>python313Packages.strawberry-django</li>
    <li>python313Packages.strawberry-django.dist</li>
    <li>python313Packages.strawberry-graphql</li>
    <li>python313Packages.strawberry-graphql.dist</li>
  </ul>
</details>

minijackson reviewed (approved):
Looks good, thanks a lot!

Only one final rebase needed, someone edited the plugins a few hours ago…

---

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 414367`
Commit: `4bfb13e0f4043757ca517ed276aa56f6983c3e5a`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>netbox_4_1</li>
    <li>netbox_4_2</li>
  </ul>
</details>
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 48 packages built:</summary>
  <ul>
    <li>netbox (netbox_4_3)</li>
    <li>nixpkgs-manual</li>
    <li>python312Packages.netbox-bgp</li>
    <li>python312Packages.netbox-bgp.dist</li>
    <li>python312Packages.netbox-documents</li>
    <li>python312Packages.netbox-documents.dist</li>
    <li>python312Packages.netbox-interface-synchronization</li>
    <li>python312Packages.netbox-interface-synchronization.dist</li>
    <li>python312Packages.netbox-plugin-prometheus-sd</li>
    <li>python312Packages.netbox-plugin-prometheus-sd.dist</li>
    <li>python312Packages.netbox-qrcode</li>
    <li>python312Packages.netbox-qrcode.dist</li>
    <li>python312Packages.netbox-reorder-rack</li>
    <li>python312Packages.netbox-reorder-rack.dist</li>
    <li>python312Packages.netbox-routing</li>
    <li>python312Packages.netbox-routing.dist</li>
    <li>python312Packages.strawberry-django</li>
    <li>python312Packages.strawberry-django.dist</li>
    <li>python312Packages.strawberry-graphql</li>
    <li>python312Packages.strawberry-graphql.dist</li>
    <li>python313Packages.netbox-attachments</li>
    <li>python313Packages.netbox-attachments.dist</li>
    <li>python313Packages.netbox-bgp</li>
    <li>python313Packages.netbox-bgp.dist</li>
    <li>python313Packages.netbox-contract</li>
    <li>python313Packages.netbox-contract.dist</li>
    <li>python313Packages.netbox-documents</li>
    <li>python313Packages.netbox-documents.dist</li>
    <li>python313Packages.netbox-floorplan-plugin</li>
    <li>python313Packages.netbox-floorplan-plugin.dist</li>
    <li>python313Packages.netbox-interface-synchronization</li>
    <li>python313Packages.netbox-interface-synchronization.dist</li>
    <li>python313Packages.netbox-napalm-plugin</li>
    <li>python313Packages.netbox-napalm-plugin.dist</li>
    <li>python313Packages.netbox-plugin-prometheus-sd</li>
    <li>python313Packages.netbox-plugin-prometheus-sd.dist</li>
    <li>python313Packages.netbox-qrcode</li>
    <li>python313Packages.netbox-qrcode.dist</li>
    <li>python313Packages.netbox-reorder-rack</li>
    <li>python313Packages.netbox-reorder-rack.dist</li>
    <li>python313Packages.netbox-routing</li>
    <li>python313Packages.netbox-routing.dist</li>
    <li>python313Packages.netbox-topology-views</li>
    <li>python313Packages.netbox-topology-views.dist</li>
    <li>python313Packages.strawberry-django</li>
    <li>python313Packages.strawberry-django.dist</li>
    <li>python313Packages.strawberry-graphql</li>
    <li>python313Packages.strawberry-graphql.dist</li>
  </ul>
</details>

minijackson reviewed (approved):
Looks good! @leona-ya can you update your review? Apparently I can't merge if a member has a "Request changes" review.

---

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 414367`
Commit: `a861a2d60ec593936873b99c97b0270910db19dc`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>netbox_4_1</li>
    <li>netbox_4_2</li>
  </ul>
</details>
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 48 packages built:</summary>
  <ul>
    <li>netbox (netbox_4_3)</li>
    <li>nixpkgs-manual</li>
    <li>python312Packages.netbox-bgp</li>
    <li>python312Packages.netbox-bgp.dist</li>
    <li>python312Packages.netbox-documents</li>
    <li>python312Packages.netbox-documents.dist</li>
    <li>python312Packages.netbox-interface-synchronization</li>
    <li>python312Packages.netbox-interface-synchronization.dist</li>
    <li>python312Packages.netbox-plugin-prometheus-sd</li>
    <li>python312Packages.netbox-plugin-prometheus-sd.dist</li>
    <li>python312Packages.netbox-qrcode</li>
    <li>python312Packages.netbox-qrcode.dist</li>
    <li>python312Packages.netbox-reorder-rack</li>
    <li>python312Packages.netbox-reorder-rack.dist</li>
    <li>python312Packages.netbox-routing</li>
    <li>python312Packages.netbox-routing.dist</li>
    <li>python312Packages.strawberry-django</li>
    <li>python312Packages.strawberry-django.dist</li>
    <li>python312Packages.strawberry-graphql</li>
    <li>python312Packages.strawberry-graphql.dist</li>
    <li>python313Packages.netbox-attachments</li>
    <li>python313Packages.netbox-attachments.dist</li>
    <li>python313Packages.netbox-bgp</li>
    <li>python313Packages.netbox-bgp.dist</li>
    <li>python313Packages.netbox-contract</li>
    <li>python313Packages.netbox-contract.dist</li>
    <li>python313Packages.netbox-documents</li>
    <li>python313Packages.netbox-documents.dist</li>
    <li>python313Packages.netbox-floorplan-plugin</li>
    <li>python313Packages.netbox-floorplan-plugin.dist</li>
    <li>python313Packages.netbox-interface-synchronization</li>
    <li>python313Packages.netbox-interface-synchronization.dist</li>
    <li>python313Packages.netbox-napalm-plugin</li>
    <li>python313Packages.netbox-napalm-plugin.dist</li>
    <li>python313Packages.netbox-plugin-prometheus-sd</li>
    <li>python313Packages.netbox-plugin-prometheus-sd.dist</li>
    <li>python313Packages.netbox-qrcode</li>
    <li>python313Packages.netbox-qrcode.dist</li>
    <li>python313Packages.netbox-reorder-rack</li>
    <li>python313Packages.netbox-reorder-rack.dist</li>
    <li>python313Packages.netbox-routing</li>
    <li>python313Packages.netbox-routing.dist</li>
    <li>python313Packages.netbox-topology-views</li>
    <li>python313Packages.netbox-topology-views.dist</li>
    <li>python313Packages.strawberry-django</li>
    <li>python313Packages.strawberry-django.dist</li>
    <li>python313Packages.strawberry-graphql</li>
    <li>python313Packages.strawberry-graphql.dist</li>
  </ul>
</details>
